src/pages/Main/Main.js

```diff
@@ -1,8 +1,86 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import Products from './Products';
+import SortingButton from './SortingButton';
 import './Main.scss';
 
 const Main = () => {
-  return <div>메인</div>;
+  const [productData, setProductData] = useState([]);
+  useEffect(() => {
+    fetch('data/productData.json')
+      .then(res => res.json())
+      .then(data => {
+        setProductData(data);
+      });
+  }, []);
+
+  const abcSort = () => {
+    let abcSorting = [...productData];
+    let abcCompare = key => (a, b) => {
+      return a[key] > b[key] ? 1 : a[key] < b[key] ? -1 : 0;
+    };
+    abcSorting.sort(abcCompare('title'));
+    setProductData(abcSorting);
+  };
+
+  const priceSort = () => {
+    let priceSorting = [...productData];
+    let priceCompare = key => (a, b) => {
+      return a[key] < b[key] ? 1 : a[key] > b[key] ? -1 : 0;
+    };
+    priceSorting.sort(priceCompare('price'));
```

Comment:
이 부분 설명해주실 수 있을까요? 해당 함수가 인자에 따라 다르게 재사용되지 않는다면 단순하게 다음과 같은 방식으로 작성해도 될 것 같습니다.
```suggestion
    const listSortedByPrice = [...productList].sort((a, b) => a.price > b.price ? 1 : -1)
```

Reply:
초기에 작성한 저 상태에서 sort를 사용할 때 key값으로 productData에 대해 인덱스 형태로 접근하려는 방식이었습니다.
해당 부분은 저도 작성하고 깔끔하지 않다고 생각되어 지속적으로 고민하고 있었는데 조언해주신 부분으로 고려해서 재 작성하겠습니다.